SourceKitStressTester/Sources/StressTester/SourceKitDocument.swift

```diff
@@ -232,60 +232,71 @@ class SourceKitDocument {
-          completionOptions.add(SourceKitdUID(string: "key.codecomplete.\(option)"), value: intValue)
-        } else {
-          completionOptions.add(SourceKitdUID(string: "key.codecomplete.\(option)"), value: value)
-        }
+    let request = SourceKitdRequest(uid: request)
+    request.addParameter(.key_SourceFile, value: args.forFile.path)
+    request.addParameter(.key_Name, value: args.forFile.path)
+    if let sourceState = sourceState {
+      request.addParameter(.key_SourceText, value: sourceState.source)
+    }
+    request.addParameter(.key_Offset, value: offset)
+    request.addParameter(.key_Line, value: location.line!)
+    request.addParameter(.key_Column, value: location.column!)
+    request.addCompilerArgs(args.sourcekitdArgs)
+
+    let completionOptions = request.addDictionaryParameter(.key_CodeCompleteOptions)
+    // Don't hide any results because otherwise checking if the expected result is contained will fail
+    completionOptions.add(.key_HideLowPriority, value: 0)
+    completionOptions.add(.key_HideByName, value: 0)
+    completionOptions.add(.key_HideUnderscores, value: 0)
+
+    // Set expected results to 200 to avoid inflating the time measurements by serialization time.
+    // To make sure that the expected result is in the results, filter by its base name.
+    completionOptions.add(.key_RequestLimit, value: 200)
+
+    for (option, value) in extraOptions {
+      if let intValue = Int(value) {
+        completionOptions.add(SourceKitdUID(string: "key.codecomplete.\(option)"), value: intValue)
+      } else {
+        completionOptions.add(SourceKitdUID(string: "key.codecomplete.\(option)"), value: value)
       }
-      return request
     }
+    return request
+  }
 
-    let request = buildCompletionRequest(request: .request_CodeCompleteOpen)
-    let info = RequestInfo.codeCompleteOpen(document: documentInfo, offset: offset,
-                                        args: args.sourcekitdArgs)
-    let (response, instructions) = try sendWithTimeoutMeasuringInstructions(request, info: info) { response in
-      if let expectedResult = expectedResult {
-        try checkExpectedCompletionResult(expectedResult, in: response, info: info)
+  func codeComplete(offset: Int, expectedResult: ExpectedResult?) throws -> (request: RequestInfo, response: SourceKitdResponse, instructions: Int, reusingASTContext: Bool) {
+    var responseToReport: SourceKitdResponse
+    let openRequest = buildCompletionRequest(request: .request_CodeCompleteOpen, offset: offset, extraOptions: extraCodeCompleteOptions)
+    let openInfo = RequestInfo.codeCompleteOpen(document: documentInfo, offset: offset,
+                                                args: args.sourcekitdArgs)
+    let (openResponse, openInstructions) = try sendWithTimeoutMeasuringInstructions(openRequest, info: openInfo)
+
+    let reusingASTContext = openResponse.value.getBool(.key_ReusingASTContext)
+
+    if let expectedResult = expectedResult {
```

Comment:
Will be interesting to see how much slower the extra request makes it.

Reply:
I would expect that the update should be pretty fast, but let’s see. I started a stress tester run via https://github.com/apple/swift/pull/37710.